Issue 43/Confirmation dialogue for logout button

emilysellers opened the pull request:
## Description
This PR implements a logout confirmation dialog so that when a user clicks the `Logout` button they are prompted to confirm that they want to logout of PASS. This prevents users from accidentally logging out via a mis-click.

## Updates
This video shows that when the user clicks `Logout`, a dialog box opens which prompts the user to confirm that they want to logout. If the user clicks `Cancel` the dialog box closes. If the user clicks `Logout` the user is logged out and rerouted to the login page.

https://user-images.githubusercontent.com/84364905/232253346-42aa37d4-36c9-4094-9b4b-8e79eef8555d.mov

## Related Issue
Closes #43.

emilysellers committed:
add confirmation dialogue for clicking logout button

emilysellers committed:
implement confirmation dialogue with dialog element

emilysellers committed:
Merge branch 'Development' of https://github.com/codeforpdx/PASS into issue-43-logout-confirmation

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
           Click the following logout button to log out of your pod:{' '}
         </label>
-        <LogoutButton onLogout={handleLogout} />
+        <button type="submit" onClick={() => setShowConfirmation(true)}>
```
On Line 31, change from type="submit" to type="button"

emilysellers replied:
Updated to "button" type.

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
+          <dialog open>
+            <p>Do you want to log out now?</p>
+            <div>
+              <LogoutButton onLogout={handleLogout}>
+                <button>Logout</button>
+              </LogoutButton>
```
From Line 38 to 40, we can use the default LogoutButton component for now. So instead, we can just use:

`<LogoutButton onLogout={handleLogout} />`

emilysellers replied:
That makes sense -- this has been updated.

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
         </label>
-        <LogoutButton onLogout={handleLogout} />
+        <button type="submit" onClick={() => setShowConfirmation(true)}>
+          Logout
```
Can change this on Line 32 to Log Out to be consistent with Log In.

emilysellers replied:
Changed to "Log Out".

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
+                <button>Logout</button>
+              </LogoutButton>
+            </div>
+            <button onClick={() => setShowConfirmation(false)}>Cancel</button>
```
Can move Line 42 inside the div with LogoutButton component:

```jsx
<div>
  <LogoutButton />
  <button>Cancel</button>
<div/>
```

emilysellers replied:
This has been moved.

emilysellers committed:
remove unused code

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
+        {showConfirmation ? (
+          <dialog open>
+            <p>Do you want to log out now?</p>
+            <div>
```
On Line 37, can do a little inline-styling with style attribute with display flex, justify content center and a gap to center the buttons in one row

emilysellers replied:
Updated -- that definitely looks better!

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
+        <button type="submit" onClick={() => setShowConfirmation(true)}>
+          Logout
+        </button>
+        {showConfirmation ? (
+          <dialog open>
```
Can combine showConfirmation state with dialog open attribute, like:

`<dialog open={showConfirmation}>`

emilysellers replied:
Updated.

leekahung commented on src/components/Login/Logout.jsx:
```diff
@@ -27,7 +28,22 @@ const Logout = () => {
+            </div>
+            <button onClick={() => setShowConfirmation(false)}>Cancel</button>
+          </dialog>
+        ) : (
+          <div></div>
+        )}
```
If we combine dialog with showConfirmation state, ternary will not be needed anymore since it'll become just a singular component:

```jsx
<dialog open={showConfirmation}>
  {/* other components here */}
</dialog>
```

Otherwise, if we're keeping it as:

```jsx
{showConfirmation
  ? <dialog open>{/* other components here */}</dialog> 
  : <div></div>
}
```

We can replace the empty div with null

emilysellers replied:
That's a great point -- I've updated this to your first suggestion.

leekahung reviewed (changes requested):
I think it looks great! (I'm quite curious about what program you used for the .gif/.mov file) I've added some feedback about the changes.

emilysellers committed:
add changes per Ka Hung's recommendations

emilysellers commented:
> I think it looks great! (I'm quite curious about what program you used for the .gif/.mov file) I've added some feedback about the changes.

Thanks for the feedback, @leekahung! I've added the changes you recommended. For the .mov file I used the built-in macOS Screenshot utility - basic but effective!

leekahung committed:
Merge branch 'Development' into issue-43-logout-confirmation

xscottxbrownx commented:
@emilysellers do you still need the `check-43` branch? or can it be deleted?

emilysellers commented:
@xscottxbrownx, yes, this branch can be deleted, if it hasn't been already.
